### ISSUE TITLE
Electrum server sync is stuck

### ISSUE THREAD

**Issue by imacfan** [mynodebtc/mynode#569]
**Describe the bug**
After upgrade to recent version 0.2.43 the electrum server sync is stuck at 26%. It's been more than 24 hours now. Tried rebooting and restart. It works for few minutes and then stops again. 
**Expected behavior**
Full sync of electrum server 

**MyNode hardware (please complete the following information):**
 - Device: [e.., Raspi4,]
 - Version [e.g. 0.2.43]

**Comment by tehelsper** [comment]
Can you share the log output of Electrum from the status page? Also, how much RAM does your device have?

**Comment by imacfan** [comment]
> Can you share the log output of Electrum from the status page? Also, how much RAM does your device have?

RAM is 8gb. 50% available. Let me find the logs

**Comment by imacfan** [comment]
> Can you share the log output of Electrum from the status page? Also, how much RAM does your device have?

Logs:

+ exit 0
Oct  5 22:03:51 myNode electrs[19795]: Config { network: Bitcoin, db_path: "/mnt/hdd/mynode/electrs/bitcoin", daemon_dir: "/home/bitcoin/.bitcoin", daemon_auth: CookieFile("/home/bitcoin/.bitcoin/.cookie"), daemon_rpc_addr: V4(127.0.0.1:8332), daemon_p2p_addr: V4(127.0.0.1:8333), electrum_rpc_addr: V4(0.0.0.0:50001), monitoring_addr: V4(127.0.0.1:4224), wait_duration: 15s, jsonrpc_timeout: 30s, index_batch_size: 12, index_lookup_limit: Some(200), auto_reindex: true, ignore_mempool: false, sync_once: false, server_banner: "Welcome to electrs 0.9.0 (Electrum Rust Server)!", args: [] }
Oct  5 22:03:51 myNode electrs[19795]: [2021-10-06T03:03:51.764Z INFO  electrs::metrics::metrics_impl] serving Prometheus metrics on 127.0.0.1:4224
Oct  5 22:03:51 myNode electrs[19795]: [2021-10-06T03:03:51.878Z INFO  electrs::db] "/mnt/hdd/mynode/electrs/bitcoin": 1451 SST files, 0.319490712 GB, 0.024368898 Grows
Oct  5 22:03:52 myNode electrs[19795]: [2021-10-06T03:03:52.985Z INFO  electrs::chain] loading 189744 headers, tip=0000000000000398cfd98d3ed76d29f27a8979dee4751b58ae98388fe31c043f
Oct  5 22:03:54 myNode electrs[19795]: [2021-10-06T03:03:54.135Z INFO  electrs::chain] chain updated: tip=0000000000000398cfd98d3ed76d29f27a8979dee4751b58ae98388fe31c043f, height=189744
Oct  5 22:03:54 myNode electrs[19795]: [2021-10-06T03:03:54.162Z INFO  electrs::index] indexing 2000 blocks: [189745..191744]
Oct  5 22:03:54 myNode electrs[19795]: [2021-10-06T03:03:54.191Z INFO  electrs::db] closing DB at /mnt/hdd/mynode/electrs/bitcoin
Oct  5 22:03:54 myNode electrs[19795]: Error: initial sync failed
Oct  5 22:03:54 myNode electrs[19795]: Caused by:
Oct  5 22:03:54 myNode electrs[19795]:     0: failed to get block 00000000000002a22f646ef9446578300ef280505bae391e65ca50611c9cb7f0
Oct  5 22:03:54 myNode electrs[19795]:     1: p2p failed to recv
Oct  5 22:03:54 myNode electrs[19795]:     2: I/O error: unexpected end of file
Oct  5 22:04:54 myNode electrs[19984]: Checking if device is shutting down...
Oct  5 22:04:54 myNode electrs[19984]: Not shutting down!
Oct  5 22:04:54 myNode electrs[19985]: + set -e
Oct  5 22:04:54 myNode electrs[19985]: + echo 'Checking if Bitcoin is synced...'
Oct  5 22:04:54 myNode electrs[19985]: Checking if Bitcoin is synced...
Oct  5 22:04:54 myNode electrs[19985]: + '[' '!' -f /mnt/hdd/mynode/.mynode_bitcoin_synced ']'
Oct  5 22:04:54 myNode electrs[19985]: + bitcoin-cli -datadir=/mnt/hdd/mynode/bitcoin -rpcwait getblockchaininfo
Oct  5 22:04:54 myNode electrs[19985]: {
Oct  5 22:04:54 myNode electrs[19985]:   "chain": "main",
Oct  5 22:04:54 myNode electrs[19985]:   "blocks": 703773,
Oct  5 22:04:54 myNode electrs[19985]:   "headers": 703773,
Oct  5 22:04:54 myNode electrs[19985]:   "bestblockhash": "00000000000000000007ccb5c145957f938f9b2990f3bbfa364912d7b03f2164",
Oct  5 22:04:54 myNode electrs[19985]:   "difficulty": 19893045048575.13,
Oct  5 22:04:54 myNode electrs[19985]:   "mediantime": 1633485084,
Oct  5 22:04:54 myNode electrs[19985]:   "verificationprogress": 0.9999977539872482,
Oct  5 22:04:54 myNode electrs[19985]:   "initialblockdownload": false,
Oct  5 22:04:54 myNode electrs[19985]:   "chainwork": "00000000000000000000000000000000000000002269dd487208e594f6c00e6e",
Oct  5 22:04:54 myNode electrs[19985]:   "size_on_disk": 417827993790,
Oct  5 22:04:54 myNode electrs[19985]:   "pruned": false,
Oct  5 22:04:54 myNode electrs[19985]:   "softforks": {
Oct  5 22:04:54 myNode electrs[19985]:     "bip34": {
Oct  5 22:04:54 myNode electrs[19985]:       "type": "buried",
Oct  5 22:04:54 myNode electrs[19985]:       "active": true,
Oct  5 22:04:54 myNode electrs[19985]:       "height": 227931
Oct  5 22:04:54 myNode electrs[19985]:     },
Oct  5 22:04:54 myNode electrs[19985]:     "bip66": {
Oct  5 22:04:54 myNode electrs[19985]:       "type": "buried",
Oct  5 22:04:54 myNode electrs[19985]:       "active": true,
Oct  5 22:04:54 myNode electrs[19985]:       "height": 363725
Oct  5 22:04:54 myNode electrs[19985]:     },
Oct  5 22:04:54 myNode electrs[19985]:     "bip65": {
Oct  5 22:04:54 myNode electrs[19985]:       "type": "buried",
Oct  5 22:04:54 myNode electrs[19985]:       "active": true,
Oct  5 22:04:54 myNode electrs[19985]:       "height": 388381
Oct  5 22:04:54 myNode electrs[19985]:     },
Oct  5 22:04:54 myNode electrs[19985]:     "csv": {
Oct  5 22:04:54 myNode electrs[19985]:       "type": "buried",
Oct  5 22:04:54 myNode electrs[19985]:       "active": true,
Oct  5 22:04:54 myNode electrs[19985]:       "height": 419328
Oct  5 22:04:54 myNode electrs[19985]:     },
Oct  5 22:04:54 myNode electrs[19985]:     "segwit": {
Oct  5 22:04:54 myNode electrs[19985]:       "type": "buried",
Oct  5 22:04:54 myNode electrs[19985]:       "active": true,
Oct  5 22:04:54 myNode electrs[19985]:       "height": 481824
Oct  5 22:04:54 myNode electrs[19985]:     },
Oct  5 22:04:54 myNode electrs[19985]:     "taproot": {
Oct  5 22:04:54 myNode electrs[19985]:       "type": "bip9",
Oct  5 22:04:54 myNode electrs[19985]:       "bip9": {
Oct  5 22:04:54 myNode electrs[19985]:         "status": "locked_in",
Oct  5 22:04:54 myNode electrs[19985]:         "start_time": 1619222400,
Oct  5 22:04:54 myNode electrs[19985]:         "timeout": 1628640000,
Oct  5 22:04:54 myNode electrs[19985]:         "since": 687456,
Oct  5 22:04:54 myNode electrs[19985]:         "min_activation_height": 709632
Oct  5 22:04:54 myNode electrs[19985]:       },
Oct  5 22:04:54 myNode electrs[19985]:       "active": false
Oct  5 22:04:54 myNode electrs[19985]:     }
Oct  5 22:04:54 myNode electrs[19985]:   },
Oct  5 22:04:54 myNode electrs[19985]:   "warnings": ""
Oct  5 22:04:54 myNode electrs[19985]: }
Oct  5 22:04:54 myNode electrs[19985]: + exit 0
Oct  5 22:04:54 myNode electrs[19987]: Config { network: Bitcoin, db_path: "/mnt/hdd/mynode/electrs/bitcoin", daemon_dir: "/home/bitcoin/.bitcoin", daemon_auth: CookieFile("/home/bitcoin/.bitcoin/.cookie"), daemon_rpc_addr: V4(127.0.0.1:8332), daemon_p2p_addr: V4(127.0.0.1:8333), electrum_rpc_addr: V4(0.0.0.0:50001), monitoring_addr: V4(127.0.0.1:4224), wait_duration: 15s, jsonrpc_timeout: 30s, index_batch_size: 12, index_lookup_limit: Some(200), auto_reindex: true, ignore_mempool: false, sync_once: false, server_banner: "Welcome to electrs 0.9.0 (Electrum Rust Server)!", args: [] }
Oct  5 22:04:54 myNode electrs[19987]: [2021-10-06T03:04:54.264Z INFO  electrs::metrics::metrics_impl] serving Prometheus metrics on 127.0.0.1:4224
Oct  5 22:04:54 myNode electrs[19987]: [2021-10-06T03:04:54.380Z INFO  electrs::db] "/mnt/hdd/mynode/electrs/bitcoin": 1452 SST files, 0.319491517 GB, 0.024368898 Grows
Oct  5 22:04:55 myNode electrs[19987]: [2021-10-06T03:04:55.560Z INFO  electrs::chain] loading 189744 headers, tip=0000000000000398cfd98d3ed76d29f27a8979dee4751b58ae98388fe31c043f
Oct  5 22:04:56 myNode electrs[19987]: [2021-10-06T03:04:56.658Z INFO  electrs::chain] chain updated: tip=0000000000000398cfd98d3ed76d29f27a8979dee4751b58ae98388fe31c043f, height=189744
Oct  5 22:04:56 myNode electrs[19987]: [2021-10-06T03:04:56.684Z INFO  electrs::index] indexing 2000 blocks: [189745..191744]
Oct  5 22:04:56 myNode electrs[19987]: [2021-10-06T03:04:56.736Z INFO  electrs::db] closing DB at /mnt/hdd/mynode/electrs/bitcoin
Oct  5 22:04:56 myNode electrs[19987]: Error: initial sync failed
Oct  5 22:04:56 myNode electrs[19987]: Caused by:
Oct  5 22:04:56 myNode electrs[19987]:     0: failed to get block 00000000000002a22f646ef9446578300ef280505bae391e65ca50611c9cb7f0
Oct  5 22:04:56 myNode electrs[19987]:     1: p2p failed to recv
Oct  5 22:04:56 myNode electrs[19987]:     2: I/O error: unexpected end of file
Oct  5 22:05:56 myNode electrs[20187]: Checking if device is shutting down...
Oct  5 22:05:56 myNode electrs[20187]: Not shutting down!
Oct  5 22:05:57 myNode electrs[20188]: + set -e
Oct  5 22:05:57 myNode electrs[20188]: + echo 'Checking if Bitcoin is synced...'
Oct  5 22:05:57 myNode electrs[20188]: Checking if Bitcoin is synced...
Oct  5 22:05:57 myNode electrs[20188]: + '[' '!' -f /mnt/hdd/mynode/.mynode_bitcoin_synced ']'
Oct  5 22:05:57 myNode electrs[20188]: + bitcoin-cli -datadir=/mnt/hdd/mynode/bitcoin -rpcwait getblockchaininfo
Oct  5 22:05:57 myNode electrs[20188]: {
Oct  5 22:05:57 myNode electrs[20188]:   "chain": "main",
Oct  5 22:05:57 myNode electrs[20188]:   "blocks": 703773,
Oct  5 22:05:57 myNode electrs[20188]:   "headers": 703773,
Oct  5 22:05:57 myNode electrs[20188]:   "bestblockhash": "00000000000000000007ccb5c145957f938f9b2990f3bbfa364912d7b03f2164",
Oct  5 22:05:57 myNode electrs[20188]:   "difficulty": 19893045048575.13,
Oct  5 22:05:57 myNode electrs[20188]:   "mediantime": 1633485084,
Oct  5 22:05:57 myNode electrs[20188]:   "verificationprogress": 0.9999975279513799,
Oct  5 22:05:57 myNode electrs[20188]:   "initialblockdownload": false,
Oct  5 22:05:57 myNode electrs[20188]:   "chainwork": "00000000000000000000000000000000000000002269dd487208e594f6c00e6e",
Oct  5 22:05:57 myNode electrs[20188]:   "size_on_disk": 417827993790,
Oct  5 22:05:57 myNode electrs[20188]:   "pruned": false,
Oct  5 22:05:57 myNode electrs[20188]:   "softforks": {
Oct  5 22:05:57 myNode electrs[20188]:     "bip34": {
Oct  5 22:05:57 myNode electrs[20188]:       "type": "buried",
Oct  5 22:05:57 myNode electrs[20188]:       "active": true,
Oct  5 22:05:57 myNode electrs[20188]:       "height": 227931
Oct  5 22:05:57 myNode electrs[20188]:     },
Oct  5 22:05:57 myNode electrs[20188]:     "bip66": {
Oct  5 22:05:57 myNode electrs[20188]:       "type": "buried",
Oct  5 22:05:57 myNode electrs[20188]:       "active": true,
Oct  5 22:05:57 myNode electrs[20188]:       "height": 363725
Oct  5 22:05:57 myNode electrs[20188]:     },
Oct  5 22:05:57 myNode electrs[20188]:     "bip65": {
Oct  5 22:05:57 myNode electrs[20188]:       "type": "buried",
Oct  5 22:05:57 myNode electrs[20188]:       "active": true,
Oct  5 22:05:57 myNode electrs[20188]:       "height": 388381
Oct  5 22:05:57 myNode electrs[20188]:     },
Oct  5 22:05:57 myNode electrs[20188]:     "csv": {
Oct  5 22:05:57 myNode electrs[20188]:       "type": "buried",
Oct  5 22:05:57 myNode electrs[20188]:       "active": true,
Oct  5 22:05:57 myNode electrs[20188]:       "height": 419328
Oct  5 22:05:57 myNode electrs[20188]:     },
Oct  5 22:05:57 myNode electrs[20188]:     "segwit": {
Oct  5 22:05:57 myNode electrs[20188]:       "type": "buried",
Oct  5 22:05:57 myNode electrs[20188]:       "active": true,
Oct  5 22:05:57 myNode electrs[20188]:       "height": 481824
Oct  5 22:05:57 myNode electrs[20188]:     },
Oct  5 22:05:57 myNode electrs[20188]:     "taproot": {
Oct  5 22:05:57 myNode electrs[20188]:       "type": "bip9",
Oct  5 22:05:57 myNode electrs[20188]:       "bip9": {
Oct  5 22:05:57 myNode electrs[20188]:         "status": "locked_in",
Oct  5 22:05:57 myNode electrs[20188]:         "start_time": 1619222400,
Oct  5 22:05:57 myNode electrs[20188]:         "timeout": 1628640000,
Oct  5 22:05:57 myNode electrs[20188]:         "since": 687456,
Oct  5 22:05:57 myNode electrs[20188]:         "min_activation_height": 709632
Oct  5 22:05:57 myNode electrs[20188]:       },
Oct  5 22:05:57 myNode electrs[20188]:       "active": false
Oct  5 22:05:57 myNode electrs[20188]:     }
Oct  5 22:05:57 myNode electrs[20188]:   },
Oct  5 22:05:57 myNode electrs[20188]:   "warnings": ""
Oct  5 22:05:57 myNode electrs[20188]: }
Oct  5 22:05:57 myNode electrs[20188]: + exit 0
Oct  5 22:05:57 myNode electrs[20190]: Config { network: Bitcoin, db_path: "/mnt/hdd/mynode/electrs/bitcoin", daemon_dir: "/home/bitcoin/.bitcoin", daemon_auth: CookieFile("/home/bitcoin/.bitcoin/.cookie"), daemon_rpc_addr: V4(127.0.0.1:8332), daemon_p2p_addr: V4(127.0.0.1:8333), electrum_rpc_addr: V4(0.0.0.0:50001), monitoring_addr: V4(127.0.0.1:4224), wait_duration: 15s, jsonrpc_timeout: 30s, index_batch_size: 12, index_lookup_limit: Some(200), auto_reindex: true, ignore_mempool: false, sync_once: false, server_banner: "Welcome to electrs 0.9.0 (Electrum Rust Server)!", args: [] }
Oct  5 22:05:57 myNode electrs[20190]: [2021-10-06T03:05:57.047Z INFO  electrs::metrics::metrics_impl] serving Prometheus metrics on 127.0.0.1:4224
Oct  5 22:05:57 myNode electrs[20190]: [2021-10-06T03:05:57.159Z INFO  electrs::db] "/mnt/hdd/mynode/electrs/bitcoin": 1453 SST files, 0.319492322 GB, 0.024368898 Grows
Oct  5 22:05:58 myNode electrs[20190]: [2021-10-06T03:05:58.214Z INFO  electrs::chain] loading 189744 headers, tip=0000000000000398cfd98d3ed76d29f27a8979dee4751b58ae98388fe31c043f
Oct  5 22:05:59 myNode electrs[20190]: [2021-10-06T03:05:59.329Z INFO  electrs::chain] chain updated: tip=0000000000000398cfd98d3ed76d29f27a8979dee4751b58ae98388fe31c043f, height=189744
Oct  5 22:05:59 myNode electrs[20190]: [2021-10-06T03:05:59.357Z INFO  electrs::index] indexing 2000 blocks: [189745..191744]
Oct  5 22:05:59 myNode electrs[20190]: [2021-10-06T03:05:59.377Z INFO  electrs::db] closing DB at /mnt/hdd/mynode/electrs/bitcoin
Oct  5 22:05:59 myNode electrs[20190]: Error: initial sync failed
Oct  5 22:05:59 myNode electrs[20190]: Caused by:
Oct  5 22:05:59 myNode electrs[20190]:     0: failed to get block 00000000000002a22f646ef9446578300ef280505bae391e65ca50611c9cb7f0
Oct  5 22:05:59 myNode electrs[20190]:     1: p2p failed to recv
Oct  5 22:05:59 myNode electrs[20190]:     2: I/O error: unexpected end of file
Oct  5 22:06:59 myNode electrs[20383]: Checking if device is shutting down...
Oct  5 22:06:59 myNode electrs[20383]: Not shutting down!
Oct  5 22:06:59 myNode electrs[20384]: + set -e
Oct  5 22:06:59 myNode electrs[20384]: + echo 'Checking if Bitcoin is synced...'
Oct  5 22:06:59 myNode electrs[20384]: Checking if Bitcoin is synced...
Oct  5 22:06:59 myNode electrs[20384]: + '[' '!' -f /mnt/hdd/mynode/.mynode_bitcoin_synced ']'
Oct  5 22:06:59 myNode electrs[20384]: + bitcoin-cli -datadir=/mnt/hdd/mynode/bitcoin -rpcwait getblockchaininfo
Oct  5 22:06:59 myNode electrs[20384]: {
Oct  5 22:06:59 myNode electrs[20384]:   "chain": "main",
Oct  5 22:06:59 myNode electrs[20384]:   "blocks": 703773,
Oct  5 22:06:59 myNode electrs[20384]:   "headers": 703773,
Oct  5 22:06:59 myNode electrs[20384]:   "bestblockhash": "00000000000000000007ccb5c145957f938f9b2990f3bbfa364912d7b03f2164",
Oct  5 22:06:59 myNode electrs[20384]:   "difficulty": 19893045048575.13,
Oct  5 22:06:59 myNode electrs[20384]:   "mediantime": 1633485084,
Oct  5 22:06:59 myNode electrs[20384]:   "verificationprogress": 0.9999973055034823,
Oct  5 22:06:59 myNode electrs[20384]:   "initialblockdownload": false,
Oct  5 22:06:59 myNode electrs[20384]:   "chainwork": "00000000000000000000000000000000000000002269dd487208e594f6c00e6e",
Oct  5 22:06:59 myNode electrs[20384]:   "size_on_disk": 417827993790,
Oct  5 22:06:59 myNode electrs[20384]:   "pruned": false,
Oct  5 22:06:59 myNode electrs[20384]:   "softforks": {
Oct  5 22:06:59 myNode electrs[20384]:     "bip34": {
Oct  5 22:06:59 myNode electrs[20384]:       "type": "buried",
Oct  5 22:06:59 myNode electrs[20384]:       "active": true,
Oct  5 22:06:59 myNode electrs[20384]:       "height": 227931
Oct  5 22:06:59 myNode electrs[20384]:     },
Oct  5 22:06:59 myNode electrs[20384]:     "bip66": {
Oct  5 22:06:59 myNode electrs[20384]:       "type": "buried",
Oct  5 22:06:59 myNode electrs[20384]:       "active": true,
Oct  5 22:06:59 myNode electrs[20384]:       "height": 363725
Oct  5 22:06:59 myNode electrs[20384]:     },
Oct  5 22:06:59 myNode electrs[20384]:     "bip65": {
Oct  5 22:06:59 myNode electrs[20384]:       "type": "buried",
Oct  5 22:06:59 myNode electrs[20384]:       "active": true,
Oct  5 22:06:59 myNode electrs[20384]:       "height": 388381
Oct  5 22:06:59 myNode electrs[20384]:     },
Oct  5 22:06:59 myNode electrs[20384]:     "csv": {
Oct  5 22:06:59 myNode electrs[20384]:       "type": "buried",
Oct  5 22:06:59 myNode electrs[20384]:       "active": true,
Oct  5 22:06:59 myNode electrs[20384]:       "height": 419328
Oct  5 22:06:59 myNode electrs[20384]:     },
Oct  5 22:06:59 myNode electrs[20384]:     "segwit": {
Oct  5 22:06:59 myNode electrs[20384]:       "type": "buried",
Oct  5 22:06:59 myNode electrs[20384]:       "active": true,
Oct  5 22:06:59 myNode electrs[20384]:       "height": 481824
Oct  5 22:06:59 myNode electrs[20384]:     },
Oct  5 22:06:59 myNode electrs[20384]:     "taproot": {
Oct  5 22:06:59 myNode electrs[20384]:       "type": "bip9",
Oct  5 22:06:59 myNode electrs[20384]:       "bip9": {
Oct  5 22:06:59 myNode electrs[20384]:         "status": "locked_in",
Oct  5 22:06:59 myNode electrs[20384]:         "start_time": 1619222400,
Oct  5 22:06:59 myNode electrs[20384]:         "timeout": 1628640000,
Oct  5 22:06:59 myNode electrs[20384]:         "since": 687456,
Oct  5 22:06:59 myNode electrs[20384]:         "min_activation_height": 709632
Oct  5 22:06:59 myNode electrs[20384]:       },
Oct  5 22:06:59 myNode electrs[20384]:       "active": false
Oct  5 22:06:59 myNode electrs[20384]:     }
Oct  5 22:06:59 myNode electrs[20384]:   },
Oct  5 22:06:59 myNode electrs[20384]:   "warnings": ""
Oct  5 22:06:59 myNode electrs[20384]: }
Oct  5 22:06:59 myNode electrs[20384]: + exit 0
Oct  5 22:06:59 myNode electrs[20386]: Config { network: Bitcoin, db_path: "/mnt/hdd/mynode/electrs/bitcoin", daemon_dir: "/home/bitcoin/.bitcoin", daemon_auth: CookieFile("/home/bitcoin/.bitcoin/.cookie"), daemon_rpc_addr: V4(127.0.0.1:8332), daemon_p2p_addr: V4(127.0.0.1:8333), electrum_rpc_addr: V4(0.0.0.0:50001), monitoring_addr: V4(127.0.0.1:4224), wait_duration: 15s, jsonrpc_timeout: 30s, index_batch_size: 12, index_lookup_limit: Some(200), auto_reindex: true, ignore_mempool: false, sync_once: false, server_banner: "Welcome to electrs 0.9.0 (Electrum Rust Server)!", args: [] }
Oct  5 22:06:59 myNode electrs[20386]: [2021-10-06T03:06:59.514Z INFO  electrs::metrics::metrics_impl] serving Prometheus metrics on 127.0.0.1:4224
Oct  5 22:06:59 myNode electrs[20386]: [2021-10-06T03:06:59.627Z INFO  electrs::db] "/mnt/hdd/mynode/electrs/bitcoin": 1454 SST files, 0.319493127 GB, 0.024368898 Grows
Oct  5 22:07:00 myNode electrs[20386]: [2021-10-06T03:07:00.716Z INFO  electrs::chain] loading 189744 headers, tip=0000000000000398cfd98d3ed76d29f27a8979dee4751b58ae98388fe31c043f
Oct  5 22:07:01 myNode electrs[20386]: [2021-10-06T03:07:01.866Z INFO  electrs::chain] chain updated: tip=0000000000000398cfd98d3ed76d29f27a8979dee4751b58ae98388fe31c043f, height=189744
Oct  5 22:07:01 myNode electrs[20386]: [2021-10-06T03:07:01.893Z INFO  electrs::index] indexing 2000 blocks: [189745..191744]
Oct  5 22:07:01 myNode electrs[20386]: [2021-10-06T03:07:01.911Z INFO  electrs::db] closing DB at /mnt/hdd/mynode/electrs/bitcoin
Oct  5 22:07:01 myNode electrs[20386]: Error: initial sync failed
Oct  5 22:07:01 myNode electrs[20386]: Caused by:
Oct  5 22:07:01 myNode electrs[20386]:     0: failed to get block 00000000000002a22f646ef9446578300ef280505bae391e65ca50611c9cb7f0
Oct  5 22:07:01 myNode electrs[20386]:     1: p2p failed to recv
Oct  5 22:07:01 myNode electrs[20386]:     2: I/O error: unexpected end of file

**Comment by tehelsper** [comment]
Do you have a custom bitcoin config? It appears the electrum can't get data from Bitcoin any more.

Oct 5 22:07:01 myNode electrs[20386]: [2021-10-06T03:07:01.911Z INFO electrs::db] closing DB at /mnt/hdd/mynode/electrs/bitcoin
Oct 5 22:07:01 myNode electrs[20386]: Error: initial sync failed
Oct 5 22:07:01 myNode electrs[20386]: Caused by:
Oct 5 22:07:01 myNode electrs[20386]: 0: failed to get block 00000000000002a22f646ef9446578300ef280505bae391e65ca50611c9cb7f0
Oct 5 22:07:01 myNode electrs[20386]: 1: p2p failed to recv
Oct 5 22:07:01 myNode electrs[20386]: 2: I/O error: unexpected end of file

**Comment by imacfan** [comment]
> Do you have a custom bitcoin config? It appears the electrum can't get data from Bitcoin any more.
> 
> Oct 5 22:07:01 myNode electrs[20386]: [2021-10-06T03:07:01.911Z INFO electrs::db] closing DB at /mnt/hdd/mynode/electrs/bitcoin Oct 5 22:07:01 myNode electrs[20386]: Error: initial sync failed Oct 5 22:07:01 myNode electrs[20386]: Caused by: Oct 5 22:07:01 myNode electrs[20386]: 0: failed to get block 00000000000002a22f646ef9446578300ef280505bae391e65ca50611c9cb7f0 Oct 5 22:07:01 myNode electrs[20386]: 1: p2p failed to recv Oct 5 22:07:01 myNode electrs[20386]: 2: I/O error: unexpected end of file

I don't remember customising it. Simply upgraded it and it synced approx 25% without any issue...
Should I try reseting the electrum data?

**Comment by tehelsper** [comment]
Yep - that's it. If you have a custom one, you won't pick up on the automated config updates and some were needed for Electrum to sync properly. Electrum was using up all your Bitcoin download bandwidth (typically limited to 1G of upload per day).

**Comment by imacfan** [comment]
> Yep - that's it. If you have a custom one, you won't pick up on the automated config updates and some were needed for Electrum to sync properly. Electrum was using up all your Bitcoin download bandwidth (typically limited to 1G of upload per day).

ok, so reseting should fix it ? right

**Comment by imacfan** [comment]
> Yeah, reset it and it will start using the myNode default config which has some settings to not limit data between Bitcoin and Electrum.

Done. Seems to be resolved now.. syncing fast now

**Comment by tehelsper** [comment]
@seberm That first log doesn't seem to show the error. Can you share more or the part where electrs shuts down / restarts?

I'll update electrs to pick up on the SIGKILL fix. Hopefully they release a v0.9.1 soon with some of the recent updates.

The only difference I see between log 1 and 2 is that the second must be using default config, which looks like it uses Bitcoin RPC to fetch block data. The default myNode config for electrs connects as a Bitcoin peer to fetch data.

**Comment by seberm** [comment]
> That first log doesn't seem to show the error. Can you share more or the part where electrs shuts down / restarts?
Unfortunately there are no more errors in the log file. That's all I can get. The electrs process is stuck, it does not shut down. It must be killed using SIGKILL due to the bug.

> I'll update electrs to pick up on the SIGKILL fix. Hopefully they release a v0.9.1 soon with some of the recent updates.

Great :), thanks.

> The only difference I see between log 1 and 2 is that the second must be using default config, which looks like it uses Bitcoin RPC to fetch block data. The default myNode config for electrs connects as a Bitcoin peer to fetch data.

I should be using the default configurations everywhere (bitcoind.conf/electrs.toml). You are right that the second log (without specifing electrs.toml) uses the default bitcoin RPC... which is working.

For me it seems like some kind of deadlock, but I do not know for sure :/.

I am attaching the strace output of the electrs process and its childs:
```
$ cp /mnt/hdd/mynode/electrs/electrs.toml ./
$ sudo -u bitcoin strace -o strace -ff /usr/bin/electrs -vvv --conf=electrs.toml
```

I think the most relevant files are `strace.25562` and maybe `strace.25596` with `strace.25603`. Maybe you will be able to find out more information than I am.

**Comment by romanz** [comment]
> Hopefully they release a v0.9.1 soon with some of the recent updates.

Will be done this week :)

**Comment by romanz** [comment]
BTW, does this bug reproduce on regtest/signet?

**Comment by seberm** [comment]
Is it somehow possible to modify the `electrs.toml` configuration permanently?

Right now it seems it's generated automatically with every mynode startup. It would be great if we could allow users to modify it, e.g. same way as with `bitcoin.conf`.

**Comment by romanz** [comment]
> Will be done this week :)

https://github.com/romanz/electrs/pull/565

**Comment by seberm** [comment]
The problem (https://github.com/romanz/electrs/issues/564) is fixed in the latest electrs version (v0.9.1 - https://github.com/romanz/electrs/pull/565).

If there is no special reason to explicitly set the `index_batch_size = 12`, I would also suggest to keep it in its default:
- https://github.com/mynodebtc/mynode/pull/580

**Comment by seberm** [comment]
Hello @tehelsper, I think this issue is now resolved and it can be closed :).
